puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -39,10 +40,14 @@ def save
     end
 
     it "should POST the catalog command as a JSON string" do
-      command_payload = subject.munge_catalog(catalog, options).to_json
-
+      command_payload = subject.munge_catalog(catalog, Time.now.utc, options)
       http.expects(:post).with do |uri, body, headers|
-        expect(body).to eq(command_payload)
+        req = JSON.parse(body)
+        actual_producer_timestamp = extract_producer_timestamp(req)
+        req.delete("producer_timestamp")
+        command_payload.delete("producer_timestamp")
+        req == command_payload &&
+          actual_producer_timestamp <= Time.now.to_i
```

Comment:
Looks like this is used a few places, is it worth/possible putting this in a util function or some such?